sql/analyzer/resolve_columns.go

```diff
@@ -1021,7 +1021,7 @@ func pushdownGroupByAliases(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Sc
 		if n.Resolved() || !ok || len(g.GroupByExprs) == 0 {
 			return n, transform.SameTree, nil
 		}
-		if !ok || len(g.GroupByExprs) == 0 {
+		if !ok || len(g.GroupByExprs) == 0 { // TODO: this never happens because of above if???
```

Comment:
What's the TODO action here?
If the "never happens" situation actually happens, how bad is it? Should we return an error or panic?

Reply:
This was just a note for myself while I was looking through the code.
If we somehow make it to this condition and it evaluates to true, we would just do what we did in the previous block.

I'm pretty sure this block can be safely removed